pyne/vta.py

```diff
@@ -0,0 +1,500 @@
+"""Purpose:
+
+This module provides a modified ray traverse method, which is modifed based on
+http://www.cse.yorku.ca/~amana/research/grid.pdf, to calculate which voxels
+intersect with a given ray (line segment).
+Several rays are used to calculate the voxels that intersect with the facet.
+A series of rays are constructed based on both the facet and the mesh grid to
+make sure all the wanted voxels will be found. Combining all the found voxels
+for each ray, we get the voxels intersecting with the facet.
+
+These voxels will be used to construct the bounding volume for source sampling.
+But currently these functions are separate from MOAB. Interface of MOAB with
+PyNE is still in need.
+
+.. moduleauthor:: X. Zhang
+"""
+from __future__ import division
+import math
+import numpy as np
+
+
+##################################
+### Facet voxel traverse tools ###
+##################################
+
+
+def _is_point_in_mesh(bounds, point):
+    """
+    Check whether a point is in the mesh.
+
+    Parameters:
+    -----------
+    bounds : list
+        Boundaries. Nested list of the boundaries along x, y, and z directions.
+        Expected format: [[x_min, ..., x_max],
+                          [y_min, ..., y_max],
+                          [z_min, ..., z_max]]
+    point : numpy array
+        An array with size of 3, represents the point to be checked.
+
+    Return:
+    -------
+    True : when point in the mesh.
+    False : when point outside the mesh.
+    """
+    if bounds[0][0] <= point[0] <= bounds[0][-1] and \
+       bounds[1][0] <= point[1] <= bounds[1][-1] and \
+       bounds[2][0] <= point[2] <= bounds[2][-1]:
+        return True
+    else:
+        return False
+
+def _find_voxel_idx_1d(bounds_1d, cor, vec_1d):
+    """
+    Find the voxel id (in a given dimension).
+
+    Parameters:
+    -----------
+    bounds_1d : list
+        Boundary of the mesh in a given direction.
+    cor : float
+        Coordinate of the point in the given direction.
+    vec_1d : float
+        Vector attribute in the given direction. Used for determining the
+        search direction.
+
+    Return:
+    -------
+    idx : int
+        Mesh index of the given direction. This value is -1 when the point not
+        in the mesh.
+    ----------
+    """
+    idx = -1
+    bounds_1d = np.array(bounds_1d)
+    if vec_1d > 0:
+        idx = np.searchsorted(bounds_1d, cor, side='right')
+    else:
+        idx = np.searchsorted(bounds_1d, cor, side='left')
+    if idx == 0 or idx == len(bounds_1d):
+    # idx could be -1, that means point outside the mesh
+        return -1
+    else:
+        return idx - 1
+
+def _calc_vec_dir(start, end):
```

Comment:
One line!, probably no need for a function since it's too simple. Even if it's used many times in many functions, it is still one line and no need to be included in a separate function. Up to you :)